src/renderer/components/detailed-view-components/DetailedViewCore.tsx

```diff
@@ -2,30 +2,94 @@
 //
 // SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
 
-import React from 'react';
+import React, { useState } from 'react';
+import { Switch } from '@blueprintjs/core';
 import { ComputeNode } from '../../../data/GraphOnChip';
 import LinkDetails from '../LinkDetails';
+import MemoryPlotRenderer from '../memory-renderer/MemoryPlotRenderer';
+import { L1RenderConfiguration, getChartData } from '../memory-renderer/PlotConfigurations';
 
 interface DetailedViewCoreRendererProps {
     node: ComputeNode;
     temporalEpoch: number;
 }
 
 const DetailedViewCoreRenderer: React.FC<DetailedViewCoreRendererProps> = ({ node, temporalEpoch }) => {
+    /**
+     * Binary buffers: These are buffers statically allocated in L1, containing NCRISC binaries,
+     * Overlay Binaries, and Trisc Binaries. Every worker core should all have these buffers
+     * (though the sizes of these buffers may differ across cores).
+     * Since these buffers are statically allocated, they have a static reserved size that may be larger
+     * than the actual consumed size. For example I could allocate 4096 bytes (reserved size) for trisc0 binaries,
+     * but the actual trisc0 binary may only be 1024 bytes large (consumed size).
+     * We also show the address that this buffer is allocated at.
+     *
+     *
+     * Data buffers: These are dynamically allocated, and therefore differ between worker cores.
+     * These buffers serve different purposes. For example relay buffers hold temporary data to be forwarded to other cores,
+     * packer stream buffers hold data packed to L1 from the packer kernel.
+     * For these buffers it's assumed that the consumed size equals the reserved size.
+     * We also show the address that this buffer is allocated at.
+     */
+
+    const MINIMAL_MEMORY_RANGE_OFFSET = 0.998;
+    const [zoomedInView, setZoomedInView] = useState(false);
+
+    // TODO: we need to find a creative way to render CONSUMED size vs SIZE - this might become interesting metrics to show
```

Comment:
Is the consumed memory contiguous? If it is all in sequence, we could add a line saying where the consumed memory ends on the graph compared to the allocated memory.

If it is not in sequence, an overlay maybe makes sense. Or a small graph/meter on the side showing from the full memory, how much is consumed, similar to the meter for op performance.